tests/ownership_dataset_test.py

```diff
@@ -176,6 +227,34 @@ def test_own_darr_copy():
     assert sc.identical(da_methdeepcopy, make_data_array()[0])
 
 
+@pytest.mark.parametrize('data_array_wrapper', (lambda k, v: {
+    k: v
+}, lambda k, v: {
+    k: v
+}.items(), lambda k, v: sc.Dataset({k: v})),
+                         ids=['dict', 'iterator', 'Dataset'])
+def test_own_dset_init(data_array_wrapper):
```

Comment:
This is testing that the various init approaches all share the underlying buffers?

Reply:
Yes. I did it only in this test because there are a lot of possible combinations...